marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -93,62 +92,83 @@ handleAction ::
   MonadAsk Env m =>
   Action -> HalogenM State Action ChildSlots Msg m Unit
 handleAction Init = do
-  mCachedWalletsJson <- liftEffect $ getItem walletsLocalStorageKey
-  for_ mCachedWalletsJson \json ->
-    for_ (runExcept $ decodeJSON json) \cachedWallets ->
-      assign _wallets cachedWallets
-  mCachedWalletJson <- liftEffect $ getItem walletLocalStorageKey
-  for_ mCachedWalletJson \json ->
-    for_ (runExcept $ decodeJSON json) \cachedWallet ->
-      assign _subState $ Right $ Play.mkInitialState cachedWallet
+  mWalletLibraryJson <- liftEffect $ getItem walletLibraryLocalStorageKey
```

Comment:
I think this deserves a runMaybeT or runExceptT in the future

Reply:
Also, we may want to do something similar to what I did here to define a subset of the state as the session info, and automatically save the session whenever an action modifies something we care about 
https://github.com/input-output-hk/plutus/pull/2624/files#diff-878b55dcf8dc2e6ad920ddb3b04a8dad589537364775a5a35534f7f29c9c83a7R859

Reply:
Ah nice, let's do that. Maybe not in this PR though.